test/test-script.py

```diff
@@ -58,31 +58,49 @@ def assert_running(unit):
 
 assert_running("spark-wallet")
 spark_auth = re.search("login=(.*)", succeed("cat /secrets/spark-wallet-login"))[1]
-machine.wait_for_open_port(9737)
-assert_matches(f"curl -s {spark_auth}@localhost:9737", "Spark")
+machine.wait_until_succeeds("ip netns exec nb-clightning nc -z 169.254.1.17 9737")
+assert_matches(f"ip netns exec nb-clightning curl -s {spark_auth}@169.254.1.17:9737", "Spark")
 
 assert_running("lightning-charge")
 charge_auth = re.search("API_TOKEN=(.*)", succeed("cat /secrets/lightning-charge-env"))[1]
-machine.wait_for_open_port(9112)
-assert_matches(f"curl -s api-token:{charge_auth}@localhost:9112/info | jq", '"id"')
+machine.wait_until_succeeds("ip netns exec nb-nanopos nc -z 169.254.1.18 9112")
+assert_matches(
+    f"ip netns exec nb-nanopos curl -s api-token:{charge_auth}@169.254.1.18:9112/info | jq", '"id"',
+)
 
 assert_running("nanopos")
-machine.wait_for_open_port(9116)
-assert_matches("curl localhost:9116", "tshirt")
+machine.wait_until_succeeds("ip netns exec nb-lightning-charge nc -z 169.254.1.19 9116")
+assert_matches("ip netns exec nb-lightning-charge curl 169.254.1.19:9116", "tshirt")
 
 assert_running("onion-chef")
 
 # FIXME: use 'wait_for_unit' because 'create-web-index' always fails during startup due
 # to incomplete unit dependencies.
 # 'create-web-index' implicitly tests 'nodeinfo'.
 machine.wait_for_unit("create-web-index")
-machine.wait_for_open_port(80)
-assert_matches("curl localhost", "nix-bitcoin")
-assert_matches("curl -L localhost/store", "tshirt")
+machine.wait_until_succeeds("ip netns exec nb-nginx nc -z localhost 80")
+assert_matches("ip netns exec nb-nginx curl localhost", "nix-bitcoin")
+assert_matches("ip netns exec nb-nginx curl -L localhost/store", "tshirt")
 
 machine.wait_until_succeeds(log_has_string("bitcoind-import-banlist", "Importing node banlist"))
 assert_no_failure("bitcoind-import-banlist")
 
+### Security tests
+
+machine.fail("ip netns exec nb-bitcoind ping -c 1 169.254.1.17")
```

Comment:
Fix tests by reformatting, setting ping deadline, and also testing successful pings.

Reply:
Fixed in a2dc3da63530a8793e1e689927b226e15e062cd1